### DISCUSSION TITLE
How to extract directory and all contents from .TAR?

### DISCUSSION THREAD

**Discussion by sbisherCWA** [icsharpcode/SharpZipLib#565] [Q&A] [answered]
Hey folks,

I am relatively new to .NET programming and have found success extracting individual files using the SharpZipLib.Tar library (all thanks to the excellent examples in the wiki). However, I need to figure out a way to extract a specific directory and all of its contents from a .TAR file, all while maintaining the folder structure of that directory. The example in the wiki seems to only cover the extraction of individual files, would anyone have a code sample that showcases the extraction of a directory from .TAR?

I can provide more information if necessary, thank you!
-Stephen

**Comment by piksel** [comment, marked as answer]
The easiest way is probably to loop over the entries as shown in
https://github.com/icsharpcode/SharpZipLib/wiki/GZip-and-Tar-Samples#-extract-from-a-tar-with-full-control

and then check the file name (in-archive path) against the directory you want, and then skip all other entries.

**Comment by sbisherCWA** [comment]
Thank you @piksel, your description helped me to solve the issue.
